File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        Set<Integer> ans = new LinkedHashSet<>(); // preserves order
        for (int num : nums) {
            ans.add(num);
        }

        int i = 0;
        for (int num : ans) {
            nums[i++] = num; // copy back into nums
        }

        return ans.size();
    }
}

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        set<int> bag;
        for(int i=0;i<nums.size();i++){
            bag.insert(nums[i]);
        }
        nums.assign(bag.begin(),bag.end());
        return bag.size();
    }
};